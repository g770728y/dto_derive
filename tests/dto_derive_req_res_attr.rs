use dto_derive::Dto;

#[derive(Debug, PartialEq)]
struct Entity {
    pub a: String,
    pub b: i32,
    pub c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(request)]
struct EntityReq {
    a: String,
    b: i32,
    c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(response)]
struct EntityRes {
    a: String,
    b: i32,
    c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(request)]
struct EntityWrongResponse {
    a: String,
    b: i32,
    c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(response)]
struct EntityWrongRequest {
    a: String,
    b: i32,
    c: bool,
}

#[test]
fn dto_request_attr() {
    let request = EntityReq {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let expected = Entity {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let actual: Entity = request.into();
    assert_eq!(expected, actual);
}

#[test]
fn dto_response_attr() {
    let entity = Entity {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let expected = EntityRes {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let actual: EntityRes = entity.into();
    assert_eq!(expected, actual);
}

#[test]
fn dto_request_attr_response_name() {
    let request = EntityWrongResponse {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let expected = Entity {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let actual: Entity = request.into();
    assert_eq!(expected, actual);
}

#[test]
fn dto_response_attr_request_name() {
    let entity = Entity {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let expected = EntityWrongRequest {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let actual: EntityWrongRequest = entity.into();
    assert_eq!(expected, actual);
}
