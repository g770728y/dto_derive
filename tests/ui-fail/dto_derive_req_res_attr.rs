extern crate dto_derive;
use dto_derive::Dto;

#[derive(Debug, PartialEq, Dto)]
#[dto(request)]
#[dto(request)] //~ ERROR duplicate `request`/`response` attribute
struct Dto1 { }

#[derive(Debug, PartialEq, Dto)]
#[dto(request)]
#[dto(response)] //~ ERROR duplicate `request`/`response` attribute
struct Dto2 { }

#[derive(Debug, PartialEq, Dto)]
#[dto(response)]
#[dto(request)] //~ ERROR duplicate `request`/`response` attribute
struct Dto3 { }

#[derive(Debug, PartialEq, Dto)]
#[dto(response)]
#[dto(response)] //~ ERROR duplicate `request`/`response` attribute
struct Dto4 { }


fn main() { }
