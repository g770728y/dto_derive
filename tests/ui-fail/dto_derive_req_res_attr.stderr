error: duplicate `request`/`response` attribute
 --> $DIR/dto_derive_req_res_attr.rs:6:6
  |
6 | #[dto(request)] //~ ERROR duplicate `request`/`response` attribute
  |      ^^^^^^^^^

error: duplicate `request`/`response` attribute
  --> $DIR/dto_derive_req_res_attr.rs:11:6
   |
11 | #[dto(response)] //~ ERROR duplicate `request`/`response` attribute
   |      ^^^^^^^^^^

error: duplicate `request`/`response` attribute
  --> $DIR/dto_derive_req_res_attr.rs:16:6
   |
16 | #[dto(request)] //~ ERROR duplicate `request`/`response` attribute
   |      ^^^^^^^^^

error: duplicate `request`/`response` attribute
  --> $DIR/dto_derive_req_res_attr.rs:21:6
   |
21 | #[dto(response)] //~ ERROR duplicate `request`/`response` attribute
   |      ^^^^^^^^^^

error: aborting due to 4 previous errors

